runtime/logging.go

```diff
@@ -58,9 +58,15 @@ func (h *LoggingHandler) loggingEnabled(level logging.Level) bool {
 }
 
 func (h *LoggingHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	cloneHeaders := r.Header.Clone()
+
+	// set the HTTP headers via a dedicated key on the parent context irrespective of logging level
+	r = r.WithContext(logging.WithHTTPRequestContext(r.Context(), &logging.HTTPRequestContext{Header: cloneHeaders}))
+
```

Comment:
It seemed appropriate to add this here given this handler is part of the all call chains.